### DISCUSSION TITLE
What is the best way to write a startup task in Adonis?

### DISCUSSION THREAD

**Discussion by arunans23** [adonisjs/core#2104] [Help]
I am using AWS Cognito as the Identity provider for a AdonisJS backend. I am currently storing some Cognito attributes like name and email in a User table in my database. When user creation API is called, I create the User in Cognito via AWS sdk and I stored the returned cognito attributes in the User database. 

During development I regularly run the migrate:refresh command. Hence the user table is now emptied out. I want to sync the cognito user pool with my User table when I run `adonis serve` so that user table will get populated with data. Is there any way to run a task during startup? I want to use the aws SDK somewhere to achieve this usecase

Thanks in advance

**Comment by thetutlage** [comment]
Assuming you are on Adonis4.1, you can make use of the ignitor hooks https://adonisjs.com/docs/4.1/ignitor#_hooks

**Comment by arunans23** [comment]
Thanks for the suggestion. I tried using the Ignitor hooks and I am unable to import the aws-sdk module inside the hooks file. It says `Error: Cannot find module 'aws-sdk`.

My hooks file.
```
const { hooks } = require('@adonisjs/ignitor')
const AWS = use('aws-sdk')

hooks.after.providersBooted(() => {
})
```

Is there any additional step that I should follow? I am using adonis 4.1

Thanks in advance.
